### ISSUE TITLE
Catch renew errors & Delete certs

### ISSUE THREAD

**Issue by prionkor** [GUI/lua-resty-auto-ssl#173]
We have a white labeled part in our app where we provide le ssl for our users. However in few cases users changes their domain or no longer using our app in those cases attempt to renew domain fails. Here is a related log:

```
Processing login.example.com
 + Checking domain name(s) of existing cert... unchanged.
 + Checking expire date of existing cert...
 + Valid till Apr  1 12:40:01 2019 GMT (Less than 30 days). Renewing!
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting authorization for login.example.com...
 + 1 pending challenge(s)
 + Deploying challenge tokens...
 + Responding to challenge for login.example.com authorization...
Invalid challenge: DOMAIN=login.example.com RESPONSE={
  "type": "http-01",
  "status": "invalid",
  "error": {
    "type": "urn:acme:error:connection",
    "detail": "dns :: DNS problem: NXDOMAIN looking up A for login.example.com",
    "status": 400
  },
  "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/-UWJLolllllllllllllllllll/14288788129",
  "token": "-GgMs_L_llllllllllllllloTb6fw1CKKZoic"
}
```

Is it possible to perhaps cache these errors and remove these domains certs from the directory so the auto ssl doesn't attempt to renew them again?

**Comment by edeis53** [comment]
Related to an issue I opened. A possible solution is here: https://github.com/GUI/lua-resty-auto-ssl/issues/150#issuecomment-404153434

**Comment by prionkor** [comment]
@edeis53 Thank you, the solution provides catching the now allowed domain. Not expired or DNS errored domains

**Comment by GUI** [comment]
I think this should hopefully be resolved in v0.13.0 between the combination of #173 and #128. Sorry for the long delay in addressing this! So I'm going to close this, but let me know if there were still any outstanding issues with the behavior in the new release.
